### PR TITLE
fuzzy_bool function used instead of ternary_sort

### PR DESCRIPTION
**Pull request by ayush1999**
Used the `fuzzy_bool` function instead of `ternary_sort`.
Fixes : #13699

### CODE REVIEW

**Comment by ayush1999** [comment]
@smichr Can you please tell me why are these tests failing?

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -218,18 +218,14 @@ def _complement(self, other):
...
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other, lambda x: fuzzy_bool(x))
             # ignore those that are contained in self
             return Union(FiniteSet(*(sifted[False])),
-                Complement(FiniteSet(*(sifted[None])), self, evaluate=False)
-                if sifted[None] else S.EmptySet)
+                Complement(FiniteSet(*(sifted[None])), self, evaluate=False))
 
-    def symmetric_difference(self, other):
+        def symmetric_difference(self, other):
```
[the remark this answers is not in the document]
Comment (ayush1999) [reply sympy/sets/sets.py]: I've done that, but majority of the tests are still failing, and I can't find the reason .Can you please help?

File: sympy/sets/sets.py [hunk 2]
```diff
@@ -218,16 +218,12 @@ def _complement(self, other):
...
         elif isinstance(other, FiniteSet):
             from sympy.utilities.iterables import sift
+            from sympy.core.logic import fuzzy_bool
 
-            def ternary_sift(el):
-                contains = self.contains(el)
-                return contains if contains in [True, False] else None
-
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other, lambda x: fuzzy_bool(x))
```
Comment (ylemkimon) [inline sympy/sets/sets.py]: You can directly pass `fuzzy_bool` function like `sifted = sift(other, fuzzy_bool)`

File: sympy/sets/sets.py [hunk 3]
```diff
@@ -218,16 +218,12 @@ def _complement(self, other):
...
             # ignore those that are contained in self
             return Union(FiniteSet(*(sifted[False])),
-                Complement(FiniteSet(*(sifted[None])), self, evaluate=False)
-                if sifted[None] else S.EmptySet)
```
Comment (ylemkimon) [inline sympy/sets/sets.py]: You still have to check for the empty set.

File: sympy/sets/sets.py (hunk 2 above)
Comment (ylemkimon) [inline sympy/sets/sets.py]: You can directly pass `fuzzy_bool` function like `sifted = sift(other, fuzzy_bool)`

**Review by ylemkimon — CHANGES_REQUESTED** [review]

File: sympy/sets/sets.py (hunk 2 above)
Comment (jksuom) [reply sympy/sets/sets.py]: This is probably not the way `fuzzy_bool` should be used. It seems to recognize 0 and 1 but no other numbers:
 ```
In [10]: fuzzy_bool(0)
Out[10]: False

In [11]: fuzzy_bool(1)
Out[11]: True

In [12]: fuzzy_bool(2)

In [13]: fuzzy_bool(3)

```

File: sympy/sets/sets.py [hunk 4]
```diff
@@ -218,12 +218,9 @@ def _complement(self, other):
 
         elif isinstance(other, FiniteSet):
             from sympy.utilities.iterables import sift
+            from sympy.core.logic import fuzzy_bool
 
-            def ternary_sift(el):
-                contains = self.contains(el)
-                return contains if contains in [True, False] else None
```
Comment (smichr) [inline sympy/sets/sets.py]: Ahh...modify 'ternary_sift' to return `fuzzy_bool(self.contains(e1))`

**Review by smichr — COMMENTED** [review]
So leave everything as it was but just use `fuzzy_bool`.

File: sympy/sets/sets.py [hunk 5]
```diff
@@ -217,17 +217,18 @@ def _complement(self, other):
             return S.EmptySet
 
         elif isinstance(other, FiniteSet):
-            from sympy.utilities.iterables import sift
```
Comment (ylemkimon) [inline sympy/sets/sets.py]: You still need to import `sift`

File: sympy/sets/sets.py [hunk 6]
```diff
@@ -217,17 +217,18 @@ def _complement(self, other):
...
+            from sympy.core.logic import fuzzy_bool
 
             def ternary_sift(el):
                 contains = self.contains(el)
-                return contains if contains in [True, False] else None
+                return fuzzy_bool(contains)
```
Comment (ylemkimon) [inline sympy/sets/sets.py]: You can create and pass a lambda function like `lambda x: fuzzy_bool(self.contains(x))`, but it's OK.

File: sympy/sets/sets.py (hunk 5 above)
Comment (ylemkimon) [inline sympy/sets/sets.py]: You still need to import sift

File: sympy/sets/sets.py [hunk 7]
```diff
@@ -217,17 +217,18 @@ def _complement(self, other):
...
 
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other, fuzzy_bool)
             # ignore those that are contained in self
             return Union(FiniteSet(*(sifted[False])),
-                Complement(FiniteSet(*(sifted[None])), self, evaluate=False)
-                if sifted[None] else S.EmptySet)
+                         Complement(
+                             FiniteSet(*(sifted[None])), self, evaluate=False)
+                         if sifted[None] else S.EmptySet)
```
Comment (ylemkimon) [inline sympy/sets/sets.py]: These lines seem to be not changed.

**Review by ylemkimon — CHANGES_REQUESTED** [review]

File: sympy/sets/sets.py [hunk 8]
```diff
@@ -217,17 +217,18 @@ def _complement(self, other):
...
-            from sympy.utilities.iterables import sift
+            from sympy.core.logic import fuzzy_bool
 
             def ternary_sift(el):
                 contains = self.contains(el)
-                return contains if contains in [True, False] else None
+                return fuzzy_bool(contains)
 
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other, fuzzy_bool)
```
Comment (smichr) [inline sympy/sets/sets.py]: `fuzzy_bool` here should again be `ternary_sift`

**Pull request closed without merging**